code/modules/vending/_vending.dm

```diff
@@ -511,9 +522,13 @@ GLOBAL_LIST_EMPTY(vending_products)
 			var/dump_path = R.product_path
 			if(!dump_path)
 				continue
-
+			if(R.amount > LAZYLEN(R.returned_products))
+				new dump_path(get_turf(src))
+			else
+				var/obj/obj_to_dump = R.returned_products[LAZYLEN(R.returned_products)] //last in, first out
```

Comment:
Seems odd to me that unmodified items are dispensed first as freebies, then through the returned products, when vending the product on purpose does the opposite.

Reply:
It's probably unintuitive but I felt like it was pretty lame to get items you can vend for free as freebies when you're risking being instantly gibbed by the vending machine by attacking it for the freebies.

Reply:
Eh fair, just put a comment in there about it so people don't think it's an oversight